### PR TITLE
Groovy extension module transformer

### PR DESCRIPTION
**Pull request by vietj**
https://issues.apache.org/jira/browse/MSHADE-241

### CODE REVIEW

File: maven-shade-plugin/src/main/java/org/apache/maven/plugins/shade/resource/GroovyResourceTransformer.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class GroovyResourceTransformer implements ResourceTransformer
+{
+
+    static final String EXT_MODULE_NAME = "META-INF/services/org.codehaus.groovy.runtime.ExtensionModule";
+    private List<String> extensionClassesList = new ArrayList<String>();
+    private List<String> staticExtensionClassesList = new ArrayList<String>();
+    String extModuleName = "no-module-name";
+    String extModuleVersion = "1.0";
```
Comment (olamy) [inline maven-shade-plugin/src/main/java/org/apache/maven/plugins/shade/resource/GroovyResourceTransformer.java:45]: Well I'm not familiar with this groovy. But should those values be configurable?

**Comment by olamy** [comment]
merged. But I simply wonder about those values?

**Comment by vietj** [comment]
a Groovy extension module is a module that provides extra method to a Java API, each module has a name and version.

when building a fatjar, it creates a single descriptors with all methods and the module name and module version don't really have anymore a meaning, so there is a default name and version and if the user wants he can override them.

does it make sense ?

**Comment by olamy** [comment]
@asfbot please #close

**Comment by olamy** [comment]
@vietj can you please close this pr as it has been merged?

**Pull request closed without merging**